src/Surface.cpp

```diff
@@ -800,19 +803,22 @@ int Circle::intersection(Point* point, double angle, Point* points) {
     else if (discr == 0) {
       xcurr = x0;
       ycurr = -b / (2*a);
-      points[num].setCoords(xcurr, ycurr);
+      zcurr = z0;
+      points[num].setCoords(xcurr, ycurr, zcurr);
+
+      /* Check that point is in same direction as angle */
       if (angle < M_PI && ycurr > y0)
         num++;
       else if (angle > M_PI && ycurr < y0)
-        num++;
-      return num;
```

Comment:
Yeah, not sure why it was there in the first place (that was a long time ago!) but this is worrisome.